tree/ntuple/v7/src/RClusterPool.cxx

```diff
@@ -67,26 +68,59 @@ ROOT::Experimental::Detail::RClusterPool::~RClusterPool()
    {
       // Controlled shutdown of the I/O thread
       std::unique_lock<std::mutex> lock(fLockWorkQueue);
-      fWorkQueue.emplace(RWorkItem());
-      fCvHasWork.notify_one();
+      fReadQueue.emplace(RReadItem());
+      fCvHasReadWork.notify_one();
    }
    fThreadIo.join();
+
+   {
+      // Controlled shutdown of the unzip thread
+      std::unique_lock<std::mutex> lock(fLockUnzipQueue);
+      fUnzipQueue.emplace(RUnzipItem());
+      fCvHasUnzipWork.notify_one();
+   }
+   fThreadUnzip.join();
 }
 
-void ROOT::Experimental::Detail::RClusterPool::ExecLoadClusters()
+void ROOT::Experimental::Detail::RClusterPool::ExecUnzipClusters()
 {
    while (true) {
-      std::vector<RWorkItem> workItems;
+      std::vector<RUnzipItem> unzipItems;
+      {
+         std::unique_lock<std::mutex> lock(fLockUnzipQueue);
+         fCvHasUnzipWork.wait(lock, [&]{ return !fUnzipQueue.empty(); });
+         while (!fUnzipQueue.empty()) {
+            unzipItems.emplace_back(std::move(fUnzipQueue.front()));
+            fUnzipQueue.pop();
+         }
+      }
+
+      for (auto &item : unzipItems) {
+         if (!item.fCluster)
```

Comment:
Can you add a comment why we can ignore subsequent `item`s from `unzipItems`? E.g. because we're guaranteed that this is the last item? Maybe even add a `#ifndef NDEBUG` assertion?